src/interfaces/IERC7540.sol

```diff
@@ -23,83 +23,100 @@ interface IERC165 {
-    /// @param _operator The address which called `requestRedeem` function
-    /// @param _sender The address which funded the `shares` of the Request (or message sender)
-    /// @param _receiver The owner of this Request
-    /// @param _requestId The RID identifier of the Request which is being received
-    /// @param _data Additional data with no specified format
-    /// @return `bytes4(keccak256("onERC7540RedeemReceived(address,address,address,uint256,bytes)"))`
-    ///  unless throwing
-    function onERC7540RedeemReceived(address _operator, address _sender, address _receiver, uint256 _requestId, bytes _data) external returns(bytes4);
+    /**
+     * @dev Handle the receipt of an deposit Request
+     *
+     * The ERC7540 smart contract calls this function on the recipient
+     * after a `requestRedeem`. This function MAY throw to revert and reject the
+     * request. Return of other than the magic value MUST result in the
+     * transaction being reverted.
+     * Inspired by https://eips.ethereum.org/EIPS/eip-721
+     *
+     * Note: the contract address is always the message sender.
+     *
+     * @param _operator The address which called `requestRedeem` function
+     * @param _sender The address which funded the `shares` of the Request (or message sender)
+     * @param _requestId The RID identifier of the Request which is being received
+     * @param _data Additional data with no specified format
+     * @return `bytes4(keccak256("onERC7540RedeemReceived(address,address,uint256,bytes)"))`
+     *  unless throwing
+     */
+    function onERC7540RedeemReceived(address _operator, address _sender, uint256 _requestId, bytes memory _data)
+        external
+        returns (bytes4);
 }
 
 interface IERC7540Deposit {
-    event DepositRequest(address indexed sender, address indexed operator, uint256 assets);
+    event DepositRequest(address indexed sender, address indexed receiver, address indexed owner, uint256 assets);
 
     /**
      * @dev Transfers assets from sender into the Vault and submits a Request for asynchronous deposit/mint.
      *
      * - MUST support ERC-20 approve / transferFrom on asset as a deposit Request flow.
      * - MUST revert if all of assets cannot be requested for deposit/mint.
-     * - sender MUST = msg.sender unless some unspecified explicit approval is given by the caller, approval of ERC7540 vault shares is NOT enough.
+     * - owner MUST be msg.sender unless some unspecified explicit approval is given by the caller,
+     *    approval of ERC-20 shares from owner to sender is NOT enough.
      *
-     * @param assets the amount of deposit assets from sender
+     * @param assets the amount of deposit assets to transfer from owner
      * @param receiver the receiver of the request who will be able to operate the request
-     * @param sender the source of the deposit assets
-     * @param bytes additional data which may be used to approve or call the receiver contract
```

Comment:
This threw warnings as the name in the nat spec didn't match the name in the method